addons/voice/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleCloudAPI.java

```diff
@@ -229,14 +230,37 @@ File synthesizeSpeech(String text, GoogleTTSVoice voice, String codec) {
         }
     }
 
+    /**
+     * Create cache entry.
+     *
+     * @param text      Converted text.
+     * @param cacheFile Cache entry file.
+     * @param audio     Byte array of the audio.
+     * @throws IOException in case of file handling exceptions
+     */
+    private void saveAudioAndTextToFile(String text, File cacheFile, byte[] audio) throws IOException {
+        FileOutputStream fos = new FileOutputStream(cacheFile);
+        fos.write(audio);
+        fos.close();
+        logger.debug("Caching audio file {}", cacheFile.getName());
+
+        // write text to file for transparency too
+        // this allows to know which contents is in which audio file
+        String txtFileName = FilenameUtils.removeExtension(cacheFile.getName()) + ".txt";
+        FileOutputStream txtFos = new FileOutputStream(new File(cacheFolder, txtFileName));
+        txtFos.write(text.getBytes(UTF_8));
```

Comment:
AFAIK these text files are used as meta data for the cached synthesized audio data. This information could then be used to regenerate or purge (specific) content. I think that won't be possible without also storing the parameters that are used in `GoogleTTSConfig.toConfigString`?

The VoiceRSS TTS addon (where I think this caching implementation originates from) doesn't have this issue. It has  no additional settings and has only one voice per locale.

Reply:
Regenerating everything would create a peak load here which I don't want. Texts will be regenerated  anyway when they are referenced... Also not changing the code for this. Ok?

Reply:
I added the config parameters and the voice name to the txt file.